src/main/java/play/dev/filewatch/FileWatchService.java

```diff
@@ -0,0 +1,140 @@
+      File targetDirectory, int pollDelayMillis, LoggerProxy logger, Boolean disableFileHashCheck) {
+    return detect(pollDelayMillis, logger, disableFileHashCheck);
+  }
+
+  /** @deprecated Renamed to {@link #detect(int, LoggerProxy)} */
+  @Deprecated(since = "3.0.0", forRemoval = true)
+  static FileWatchService defaultWatchService(int pollDelayMillis, LoggerProxy logger) {
+    return detect(pollDelayMillis, logger, false);
+  }
+
+  /** @deprecated Renamed to {@link #detect(int, LoggerProxy, Boolean)} */
+  @Deprecated(since = "3.0.0", forRemoval = true)
+  static FileWatchService defaultWatchService(
+      int pollDelayMillis, LoggerProxy logger, Boolean disableFileHashCheck) {
+    return detect(pollDelayMillis, logger, disableFileHashCheck);
+  }
+
+  static FileWatchService detect(int pollDelayMillis, LoggerProxy logger) {
+    return detect(pollDelayMillis, logger, false);
+  }
+
+  static FileWatchService detect(
+      int pollDelayMillis, LoggerProxy logger, Boolean disableFileHashCheck) {
+    FileWatchService watchService;
+    switch (OS.getCurrent()) {
+        // If Windows or Linux, use JDK7 Watch Service (assume JDK7+)
+      case Windows:
+      case Linux:
+        {
+          watchService = jdk7(logger, disableFileHashCheck);
+          break;
+        }
+        // If macOS, use the mac implementation
+      case Mac:
+        {
+          try {
+            watchService = mac(logger, disableFileHashCheck);
+          } catch (Throwable e) {
+            logger.warn(() -> "Error loading Mac OS X watch service: " + e.getMessage());
+            logger.trace(() -> e);
+            watchService = polling(pollDelayMillis);
+          }
+          break;
+        }
+        // Fall back to polling watch service
+      default:
+        watchService = polling(pollDelayMillis);
+        break;
+    }
+    final FileWatchService delegate = watchService;
+
+    return new FileWatchService() {
+
+      @Override
+      public FileWatcher watch(Iterable<File> filesToWatch, Runnable onChange) {
+        return delegate.watch(filesToWatch, onChange);
+      }
+
+      @Override
+      public String toString() {
+        return delegate.toString();
+      }
+    };
+  }
+
+  static FileWatchService jdk7(LoggerProxy logger, boolean disableFileHashCheck) {
+    return defaultFileWatchService(logger, false, disableFileHashCheck);
+  }
+
+  static FileWatchService jdk7(LoggerProxy logger) {
+    return jdk7(logger, false);
+  }
+
+  static FileWatchService mac(LoggerProxy logger, boolean disableFileHashCheck) {
+    return defaultFileWatchService(logger, true, disableFileHashCheck);
+  }
+
+  static FileWatchService mac(LoggerProxy logger) {
+    return mac(logger, false);
+  }
+
+  static FileWatchService polling(int pollDelayMillis) {
+    return new PollingFileWatchService(pollDelayMillis);
+  }
+
+  /** This method was previously named "default" */
+  static FileWatchService defaultFileWatchService(
```

Comment:
I totally did not like the method name `def`. So I was thinking what to do here:
So we have to remove the methods `default` anyway of course. So I was thinking we previously had the methods `defaultWatchService` above and `default`. Confusing. So actually I checked what the methods do, and I think calling `defaultWatchService` now `detect` makes much more sense, because it detects the file watch service that should be used. And `default` was called `default` because it returned an instance of `DefaultFileWatchService`...  But it can be measleading IMHO, so I think calling it `defaultFileWatchService` makes more sense (specially if we have `detect` now).
I think these methods name are less confusing now.